File: plugins/fr.cea.nabla.javalib/src/linearalgebrajava/Vector.java
/*******************************************************************************
 * Copyright (c) 2020 CEA
 * This program and the accompanying materials are made available under the 
 * terms of the Eclipse Public License 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0.
 *
 * SPDX-License-Identifier: EPL-2.0
 * Contributors: see AUTHORS file
 *******************************************************************************/
package linearalgebrajava;

import java.util.Arrays;

import org.apache.commons.math3.linear.ArrayRealVector;
import org.apache.commons.math3.linear.RealVector;
//import org.apache.commons.math3.linear.OpenMapRealVector;

public class Vector
{
	private final Object lock = new Object();
	private final RealVector data;
	private final String name;

	public Vector(final String name, final int size)
	{
		this.name = name;
		// Dense
		this.data = new ArrayRealVector(size);
		// Sparse
		// this.data = new OpenMapRealVector(size);
	}

	public Vector(final String name, final RealVector data)
	{
		this.name = name;
		this.data = data;
	}

	public String getName()
	{
		return name;
	}

	public RealVector getData()
	{
		return data;
	}

	public int getSize()
	{
		return data.getDimension();
	}

	public double getValue(int i)
	{
		return data.getEntry(i);
	}

	public void setValue(int i, double value)
	{
		synchronized(lock) { data.setEntry(i, value); }
	}

	@Override
	public String toString()
	{
		return Arrays.toString(data.toArray());
	}
}

File: plugins/fr.cea.nabla.pythonlib/src/linearalgebra/linearalgebranumpy/vector.py
"""
 *******************************************************************************/
 * Copyright (c) 2022 CEA
 * This program and the accompanying materials are made available under the 
 * terms of the Eclipse Public License 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0.
 *
 * SPDX-License-Identifier: EPL-2.0
 * Contributors: see AUTHORS file
 *******************************************************************************/
"""
import numpy as np

class Vector:
	def __init__(self, name, data):
		self.__name = name
		self.__data = data

	@classmethod
	def zeros(cls, name, size):
		return cls(name, np.zeros(size, dtype=np.double))

	def getName(self):
		return self.__name

	def getSize(self):
		return len(self.__data)

	def getValue(self, i):
		return self.__data[i]

	def setValue(self, i, value):
		self.__data[i] = value

	def getData(self):
		return self.__data
